src/app/clusters/electrical-energy-measurement-server/ElectricalEnergyMeasurementCluster.h

```diff
@@ -32,31 +33,104 @@ enum class OptionalAttributes : uint32_t
-        mFeature(aFeature), mOptionalAttrs(aOptionalAttrs)
+    // Type aliases for shorter struct type names
+    using MeasurementAccuracyStruct   = ElectricalEnergyMeasurement::Structs::MeasurementAccuracyStruct::Type;
+    using EnergyMeasurementStruct     = ElectricalEnergyMeasurement::Structs::EnergyMeasurementStruct::Type;
+    using CumulativeEnergyResetStruct = ElectricalEnergyMeasurement::Structs::CumulativeEnergyResetStruct::Type;
+
+    using OptionalAttributesSet = OptionalAttributeSet<                        //
+        ElectricalEnergyMeasurement::Attributes::CumulativeEnergyImported::Id, //
+        ElectricalEnergyMeasurement::Attributes::CumulativeEnergyExported::Id, //
+        ElectricalEnergyMeasurement::Attributes::PeriodicEnergyImported::Id,   //
+        ElectricalEnergyMeasurement::Attributes::PeriodicEnergyExported::Id,   //
+        ElectricalEnergyMeasurement::Attributes::CumulativeEnergyReset::Id     //
+        >;
+
+    struct Config
+    {
+        Config(EndpointId endpoint, BitMask<ElectricalEnergyMeasurement::Feature> featureFlags,
+               BitMask<ElectricalEnergyMeasurement::OptionalAttributes> optionalAttributes) :
+            endpointId(endpoint),
+            mFeatureFlags(featureFlags)
+        {
+            mEnabledOptionalAttributes.Set<ElectricalEnergyMeasurement::Attributes::CumulativeEnergyImported::Id>(
+                featureFlags.HasAll(ElectricalEnergyMeasurement::Feature::kCumulativeEnergy,
+                                    ElectricalEnergyMeasurement::Feature::kImportedEnergy));
+            mEnabledOptionalAttributes.Set<ElectricalEnergyMeasurement::Attributes::CumulativeEnergyExported::Id>(
+                (featureFlags.HasAll(ElectricalEnergyMeasurement::Feature::kCumulativeEnergy,
+                                     ElectricalEnergyMeasurement::Feature::kExportedEnergy)));
+            mEnabledOptionalAttributes.Set<ElectricalEnergyMeasurement::Attributes::PeriodicEnergyImported::Id>(featureFlags.HasAll(
+                ElectricalEnergyMeasurement::Feature::kPeriodicEnergy, ElectricalEnergyMeasurement::Feature::kImportedEnergy));
+            mEnabledOptionalAttributes.Set<ElectricalEnergyMeasurement::Attributes::PeriodicEnergyExported::Id>(featureFlags.HasAll(
+                ElectricalEnergyMeasurement::Feature::kPeriodicEnergy, ElectricalEnergyMeasurement::Feature::kExportedEnergy));
+            mEnabledOptionalAttributes.Set<ElectricalEnergyMeasurement::Attributes::CumulativeEnergyReset::Id>(
+                optionalAttributes.Has(ElectricalEnergyMeasurement::OptionalAttributes::kOptionalAttributeCumulativeEnergyReset) &&
+                featureFlags.Has(ElectricalEnergyMeasurement::Feature::kCumulativeEnergy));
+        }
+
+        EndpointId endpointId;
+        const BitFlags<ElectricalEnergyMeasurement::Feature> mFeatureFlags;
+        OptionalAttributesSet mEnabledOptionalAttributes;
```

Comment:
should both endpoint and this optional attribute be const too? it seems for now we just want a nice constructor.

I would maybe consider not having the config provide a constructor - it is more readable to have `Foo { .a = 1, .b =2}` than `Foo(1,2)`.

Reply:
But then we would have the same problem on the EnergyMeasurementCluster object.

Reply:
nvm, figured it out.